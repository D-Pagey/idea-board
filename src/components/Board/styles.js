import styled from 'styled-components';

export const BoardWrapper = styled.div`
  align-items: center;
  display: flex;
  flex-direction: column;
`;

export const AddIcon = styled.i.attrs({
  className: 'material-icons',
  role: 'button',
  tabIndex: '0',
})`
  font-size: 30px;
  margin-top: 5px;
`;
