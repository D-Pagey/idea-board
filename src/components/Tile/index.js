import React from 'react';
import styled from 'styled-components';

const TileWrapper = styled.div`
  border: 1px solid black;
  border-radius: 6px;
  display: flex;
  flex-direction: column;
  margin: 15px 10px;
  padding: 3px;
`;

const TitleInput = styled.input.attrs({
  type: 'text',
  placeholder: 'Title',
})`
  color: blue;
  font-size: 1.5rem;
  padding: 2px;
  border-radius: 6px;
  border: 0;
`;

const TileDescription = styled.input.attrs({
  type: 'text',
  placeholder: 'Start noting your idea...',
})`
  font-size: 1rem;
  padding: 0;
`;

const DateStamp = styled.p`
  color: darkgray;
  font-size: .7rem;
  margin: 5px 2px 0 ;
`;

export default function Tile({ data }) {
  return (
    <TileWrapper>
      <TitleInput
        defaultValue={data.title}
      />
      <TileDescription
        defaultValue={data.description}
      />
      <DateStamp>
        Created at {data.date}
      </DateStamp>
    </TileWrapper>
  );
}
