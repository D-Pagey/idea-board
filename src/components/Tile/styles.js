import styled from 'styled-components';

export const TileWrapper = styled.div`
  background-color: white;
  border: 1px solid darkgray;
  border-radius: 6px;
  box-shadow: 0 10px 20px rgba(0,0,0,0.19), 0 6px 6px rgba(0,0,0,0.23);
  display: flex;
  flex-direction: column;
  margin: 15px 10px;
  padding: 4px 7px;
`;

export const TileHeader = styled.div`
  display: flex;
  justify-content: space-between;
  width: 100%;
`;

export const TitleInput = styled.input.attrs({
  type: 'text',
  placeholder: 'Title',
  name: 'title',
})`
  color: black;
  font-size: 1.5rem;
  padding: 2px;
  border-radius: 6px;
  border: 0;
  width: 75%;
`;

export const DeleteIcon = styled.i.attrs({
  className: 'material-icons',
  role: 'button',
  tabIndex: '0',
})`
  position: relative;
  top: 3px;
`;

export const TileDescription = styled.textarea.attrs({
  type: 'text',
  placeholder: 'Start noting your idea...',
  maxLength: 140,
  name: 'description',
})`
  font-family: 'Montserrat';
  font-size: .9rem;
  margin: 4px 0;
  padding: 3px 2px;
`;

export const SubtleDiv = styled.div`
  display: flex;
  justify-content: space-between;
`;

export const DateStamp = styled.p`
  color: darkgray;
  font-size: .7rem;
  margin: 5px 2px 0;
`;

export const CharacterCount = styled.p`
  color: darkgray;
  font-size: .7rem;
  margin: 5px 2px 0;
`;
