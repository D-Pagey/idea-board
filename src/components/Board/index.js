import React from 'react';
import { arrayOf, any, func } from 'prop-types';

import { BoardWrapper } from './styles';
import Tile from '../Tile';

export default function Board({ tiles, deleteTile, addTile }) {
  const renderTiles = tiles
    .map(tile => <Tile data={tile} key={tile.id} deleteTile={deleteTile} />);

  return (
    <BoardWrapper>
      {renderTiles}
      <i
        className="material-icons"
        onClick={addTile}
        role="button"
        tabIndex="0"
      >add_circle_outline
      </i>
    </BoardWrapper>
  );
}

Board.propTypes = {
  tiles: arrayOf(any),
  deleteTile: func.isRequired,
  addTile: func.isRequired,
};

Board.defaultProps = {
  tiles: [],
};
