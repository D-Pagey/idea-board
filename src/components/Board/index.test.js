import React from 'react';
import { shallow } from 'enzyme';
import each from 'jest-each';

import Tile from '../Tile';
import Board from '.';

const props = {
  tiles: [{
    date: '11:00',
    description: 'BNTI gaming worldwide',
    id: 12345,
    title: 'Nav',
  }],
  deleteTile: () => {},
  handleChange: () => {},
};

describe('Board component', () => {
  it('should render', () => {
    // create / execute
    const wrapper = shallow(<Board {...props} />);
    // assert
    expect(wrapper).toMatchSnapshot();
  });

  each([0, 2, 5, 10]).test('should render with %d tiles', (tileCount) => {
    // mock
    const tiles = Array(tileCount).fill(props.tiles[0]);
    // create / execute
    const wrapper = shallow(<Board {...props} tiles={tiles} />);
    // assert
    expect(wrapper.find(Tile)).toHaveLength(tileCount);
  });

  it('should handle delete tile', () => {
    // mock
    const mock = jest.fn();
    // create
    const wrapper = shallow(<Board {...props} deleteTile={mock} />);
    // execute
    wrapper.find(Tile).props().deleteTile();
    // assert
    expect(mock).toHaveBeenCalledWith(props.tiles[0].id);
  });

  it('should handle change tile', () => {
    // mock
    const tiles = Array(2).fill(props.tiles[0]);
    const mock = jest.fn();
    const mockEvent = {};
    // create
    const wrapper = shallow(<Board {...props} tiles={tiles} handleChange={mock} />);

    wrapper.find(Tile).forEach((node, index) => {
      // execute
      node.props().handleChange(mockEvent);
      // assert
      expect(mock).toHaveBeenLastCalledWith(index, mockEvent);
    });
  });
});
