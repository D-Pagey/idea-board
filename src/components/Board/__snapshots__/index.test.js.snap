// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Board component should render 1`] = `
<styled.div>
  <Tile
    date="11:00"
    deleteTile={[Function]}
    description="BNTI gaming worldwide"
    handleChange={[Function]}
    id={12345}
    key="12345"
    title="Nav"
    updated=""
  />
</styled.div>
`;
